shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/MySQLAdminExecutorCreator.java

```diff
@@ -159,14 +159,12 @@ private boolean isQuerySpecialSchema(final SelectStatement sqlStatement, final S
     
     private DatabaseAdminExecutor mockExecutor(final String schemaName, final SelectStatement sqlStatement, final String sql) {
         boolean isNotUseSchema = !Optional.ofNullable(schemaName).isPresent() && sqlStatement.getFrom() == null;
+        if (!hasDatabases() || !hasResources()) {
+            return new NoResourceShowExecutor(sqlStatement);
+        }
+        String driverType = ProxyContext.getInstance().getContextManager().getMetaDataContexts().getProps().getValue(ConfigurationPropertyKey.PROXY_BACKEND_DRIVER_TYPE);
```

Comment:
Getting driver type can be moved into `if (isNotUseSchema)` block to reduce unnecessary overhead since `isNotUseSchema` is not always `true`.

Reply:
I will adjust it in the next pr